source/jsonifiable.d.ts

```diff
@@ -0,0 +1,32 @@
+import type {JsonPrimitive, JsonValue} from './basic';
+
+type JsonifiableObject = {[Key in string]?: Jsonifiable} | {toJSON: () => JsonValue};
+type JsonifiableArray = Jsonifiable[];
+
+/**
+Matches a value that can be losslessly converted to JSON.
+
```

Comment:
Would you be useful to mention some use-cases for this type. (I realize there is an example, but nothing beats just reading some text quickly)